src/main/java/io/vavr/collection/Traversable.java

```diff
@@ -1704,6 +1704,15 @@ default Number sum() {
      */
     <U> Traversable<U> zipWithIndex(BiFunction<? super T, ? super Integer, ? extends U> mapper);
 
+
+    default <K, V> Map<K, V> toMap(
+            Function<? super T, K> keyMapper,
+            Function<? super T, V> valueMapper,
+            BiFunction<? super V, ? super V, ? extends V> merge) {
+        return this
```

Comment:
The functions `groupBy`, `mapValues` and `reduce` already check that the arguments are non-null but I would add explicit checks like you did before:

```java
Objects.requireNonNull(keyMapper, "keyMapper is null");
Objects.requireNonNull(valueMapper, "valueMapper is null");
Objects.requireNonNull(merge, "merge is null");
```

That way error messages contain the correct parameter name.

Please also add the following Javadoc:

```java
    /**
     * Converts this collection to a {@link Map}.
     *
     * @param keyMapper   A function that maps an element to a key
     * @param valueMapper A function that maps an element to a value
     * @param merge A function that merges values that are associated with the same key
     * @param <K>         The key type
     * @param <V>         The value type
     * @return A new {@link HashMap}.
     */
```

Reply:
Done. Thanks!